src/lsm6.ts

```diff
@@ -0,0 +1,305 @@
+import I2CPromisifiedBus from "./i2c/i2c-connection";
+
+enum RegAddr {
+    FUNC_CFG_ACCESS   = 0x01,
+
+    FIFO_CTRL1        = 0x06,
+    FIFO_CTRL2        = 0x07,
+    FIFO_CTRL3        = 0x08,
+    FIFO_CTRL4        = 0x09,
+    FIFO_CTRL5        = 0x0A,
+    ORIENT_CFG_G      = 0x0B,
+
+    INT1_CTRL         = 0x0D,
+    INT2_CTRL         = 0x0E,
+    WHO_AM_I          = 0x0F,
+    CTRL1_XL          = 0x10,
+    CTRL2_G           = 0x11,
+    CTRL3_C           = 0x12,
+    CTRL4_C           = 0x13,
+    CTRL5_C           = 0x14,
+    CTRL6_C           = 0x15,
+    CTRL7_G           = 0x16,
+    CTRL8_XL          = 0x17,
+    CTRL9_XL          = 0x18,
+    CTRL10_C          = 0x19,
+
+    WAKE_UP_SRC       = 0x1B,
+    TAP_SRC           = 0x1C,
+    D6D_SRC           = 0x1D,
+    STATUS_REG        = 0x1E,
+
+    OUT_TEMP_L        = 0x20,
+    OUT_TEMP_H        = 0x21,
+    OUTX_L_G          = 0x22,
+    OUTX_H_G          = 0x23,
+    OUTY_L_G          = 0x24,
+    OUTY_H_G          = 0x25,
+    OUTZ_L_G          = 0x26,
+    OUTZ_H_G          = 0x27,
+    OUTX_L_XL         = 0x28,
+    OUTX_H_XL         = 0x29,
+    OUTY_L_XL         = 0x2A,
+    OUTY_H_XL         = 0x2B,
+    OUTZ_L_XL         = 0x2C,
+    OUTZ_H_XL         = 0x2D,
+
+    FIFO_STATUS1      = 0x3A,
+    FIFO_STATUS2      = 0x3B,
+    FIFO_STATUS3      = 0x3C,
+    FIFO_STATUS4      = 0x3D,
+    FIFO_DATA_OUT_L   = 0x3E,
+    FIFO_DATA_OUT_H   = 0x3F,
+    TIMESTAMP0_REG    = 0x40,
+    TIMESTAMP1_REG    = 0x41,
+    TIMESTAMP2_REG    = 0x42,
+
+    STEP_TIMESTAMP_L  = 0x49,
+    STEP_TIMESTAMP_H  = 0x4A,
+    STEP_COUNTER_L    = 0x4B,
+    STEP_COUNTER_H    = 0x4C,
+
+    FUNC_SRC          = 0x53,
+
+    TAP_CFG           = 0x58,
+    TAP_THS_6D        = 0x59,
+    INT_DUR2          = 0x5A,
+    WAKE_UP_THS       = 0x5B,
+    WAKE_UP_DUR       = 0x5C,
+    FREE_FALL         = 0x5D,
+    MD1_CFG           = 0x5E,
+    MD2_CFG           = 0x5F,
+}
+
+const DS33_WHO_ID = 0x69;
+
+// Range scales for the accelerometer
+export enum AccelerometerScale {
+    SCALE_2G = "SCALE_2G",
+    SCALE_4G = "SCALE_4G",
+    SCALE_8G = "SCALE_8G",
+    SCALE_16G = "SCALE_16G"
+}
+
+// Range scales for the gyro
+export enum GyroScale {
+    SCALE_125_DPS = "SCALE_125_DPS",
+    SCALE_250_DPS = "SCALE_250_DPS",
+    SCALE_500_DPS = "SCALE_500_DPS",
+    SCALE_1000_DPS = "SCALE_1000_DPS",
+    SCALE_2000_DPS = "SCALE_2000_DPS"
+}
+
+export interface Vector3 {
+    x: number;
+    y: number;
+    z: number;
+}
+
+const ACCEL_SCALE_CTRL_BYTE: Map<AccelerometerScale, number> = new Map<AccelerometerScale, number>();
+ACCEL_SCALE_CTRL_BYTE.set(AccelerometerScale.SCALE_2G, 0x80);
```

Comment:
I can't seem to find this in the spec sheet. Where are the control bytes set for g scale? I am assuming CTRL1_XL and but that would seem to me to be 2g 0x00, 4g 0x08, 8g 0x0C, 16 0x04. Where does the 8 come in? Is that the output performance set to ODR_XL3, 1.66 kHz?

Reply:
Yep, it's setting the output performance. We're just going to lock that at a specific value and only alter the output scale